src/server.ts

```diff
@@ -32,6 +33,331 @@ app.get("/health-check", async (req, res) => {
+app.post("/users", async (req, res) => {
+  const user_name = req.body.user_name;
+  const user_isfaculty = req.body.user_isfaculty;
+
+  const text = `INSERT INTO users(user_name, user_isfaculty)  VALUES($1,$2)`;
+
+  const values = [user_name, user_isfaculty];
+
+  const postData = await client.query(text, values);
+
+  res.status(201).json(postData.rows);
+});
+
+//-----------------------------------------------------------------------------------------------------DELETE request to DATABASE by user_id
+app.delete<{ user_id: number }>("/users/:user_id", async (req, res) => {
+  const delete_user = req.params.user_id;
+  if (delete_user === undefined) {
+    res.status(404).json(delete_user);
+  } else {
+    await client.query(`DELETE FROM comments WHERE user_id = ${delete_user}`);
+    await client.query(`DELETE FROM resources WHERE user_id = ${delete_user}`);
+    await client.query(`DELETE FROM users WHERE user_id = ${delete_user}`);
+
+    res.status(200).json(delete_user);
+  }
+});
+
+//-----------------------------------------------------------------------------------------------------PATCH request to DATABASE for user_name by user_id
+app.patch<{ user_id: number }>("/users/:user_id", async (req, res) => {
+  const patch_user = req.params.user_id;
+  if (patch_user === undefined) {
+    res.status(404).json(patch_user);
+  } else {
+    const new_user_name = req.body.new_user_name;
+
+    await client.query(
+      `UPDATE users SET user_name = '${new_user_name}'  WHERE user_id = ${patch_user}`
+    );
+
+    res.status(200).json(patch_user);
+  }
+});
+
+//-----------------------------------------------------------------------------------------------------requests to DATABASE for RESOURCES table
+//-----------------------------------------------------------------------------------------------------GET request to DATABASE for ALL resources
+app.get("/resources", async (req, res) => {
+  const resources = await client.query(
+    "SELECT * FROM resources ORDER BY resource_post_date DESC"
+  );
+  res.status(200).json(resources.rows);
+});
+
+//-----------------------------------------------------------------------------------------------------GET request to DATABASE for resource by resource_id
+app.get<{ resource_id: number }>(
+  "/resources/:resource_id",
+  async (req, res) => {
+    const resource_id = req.params.resource_id;
+
+    if (resource_id === undefined) {
+      res.status(404).json(resource_id);
+    } else {
+      const resource = await client.query(
+        `SELECT * FROM resources WHERE resource_id = ${resource_id} ORDER BY resource_post_date`
+      );
+
+      res.status(200).json(resource.rows);
+    }
+  }
+);
+
+//-----------------------------------------------------------------------------------------------------POST request to DATABASE for resource with BODY
+app.post("/resources", async (req, res) => {
+  console.log(req.body);
+  const resource_name = req.body.resource_name;
```

Comment:
Could have just destructed the object as the names used are all the same e.g.

`const { resource_name, user_id, etc... } = req.body`